demos/common/cpp/models/src/hpe_model_hrnet.cpp

```diff
@@ -0,0 +1,237 @@
+    int targetSize, float confidenceThreshold) :
+    ModelBase(modelFileName),
+    aspectRatio(aspectRatio),
+    targetSize(targetSize),
+    confidenceThreshold(confidenceThreshold) {
+}
+
+void HpeHRNet::prepareInputsOutputs(CNNNetwork& cnnNetwork) {
+    // --------------------------- Configure input & output -------------------------------------------------
+    // --------------------------- Prepare input blobs ------------------------------------------------------
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    if (inputShapes.size() != 1)
+        throw std::runtime_error("Demo supports topologies only with 1 input");
+    inputsNames.push_back(inputShapes.begin()->first);
+    SizeVector& inSizeVector = inputShapes.begin()->second;
+    if (inSizeVector.size() != 4 || inSizeVector[0] != 1 || inSizeVector[1] != 3)
+        throw std::runtime_error("3-channel 4-dimensional model's input is expected");
+    InputInfo& inputInfo = *cnnNetwork.getInputsInfo().begin()->second;
+    inputInfo.setPrecision(Precision::U8);
+    inputInfo.getInputData()->setLayout(Layout::NHWC);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    const OutputsDataMap& outputInfo = cnnNetwork.getOutputsInfo();
+    if (outputInfo.size() != 2 && outputInfo.size() != 3)
+        throw std::runtime_error("Demo supports topologies only with 2 or 3 outputs");
+    for (const auto& outputLayer: outputInfo) {
+        outputLayer.second->setPrecision(Precision::FP32);
+        outputsNames.push_back(outputLayer.first);
+        const SizeVector& outputLayerDims = outputLayer.second->getTensorDesc().getDims();
+        if (outputLayerDims.size() != 4)
+                throw std::runtime_error("output layers are expected to be 4-dimensional");
+        if (outputLayerDims[0] != 1 || outputLayerDims[1] != 17)
+                throw std::runtime_error("output layers are expected to have 1 batch size and 17 channels");
+    }
+    embeddingsBlobName = findLayerByName("embeddings", outputsNames);
+    heatmapsBlobName = findLayerByName("heatmaps", outputsNames);
+    try {
+        nmsHeatmapsBlobName = findLayerByName("nms_heatmaps", outputsNames);
+    }
+    catch (const std::runtime_error&) {
+        nmsHeatmapsBlobName = heatmapsBlobName;
+    }
+}
+
+void HpeHRNet::reshape(CNNNetwork& cnnNetwork) {
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    SizeVector& inputDims = inputShapes.begin()->second;
+    if (!targetSize) {
+        targetSize =  static_cast<int>(std::min(inputDims[2], inputDims[3]));
+    }
+    int inputHeight = static_cast<int>(std::round(targetSize * aspectRatio));
+    int inputWidth = targetSize;
+    if (aspectRatio >= 1.0) {
+        std::swap(inputHeight, inputWidth);
+    } else {
+        inputHeight = static_cast<int>(std::round(targetSize / aspectRatio));
+    }
```

Comment:
Please, fix that for `HpeAssociativeEmbedding::reshape`.
May be just
```suggestion
    int inputHeight = aspectRatio >= 1.0 ? targetSize : static_cast<int>(std::round(targetSize / aspectRatio));
    int inputWidth = aspectRatio >= 1.0 ? static_cast<int>(std::round(targetSize * aspectRatio)) : targetSize;
```

Reply:
Done